### PR TITLE
Warnings

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- Fix gcc compiler warnings

ENDRELEASENOTES

### CODE REVIEW

File: include/ConformalTracking.h [hunk 1]
```diff
@@ -68,7 +70,7 @@ class ConformalTracking : public Processor {
 
   // Cell creation
   KDCluster* extrapolateCell(Cell*, double);
-  void       extendSeedCells(std::vector<Cell*>&, KDTree*, bool, std::vector<KDCluster*>);
+  void       extendSeedCells(std::vector<Cell*>&, KDTree*, bool, const std::vector<KDCluster*>&);
```
Comment (dhynds) [inline include/ConformalTracking.h]: Just to check, this just presents a copy of the vector being made, no? It still doesn't allow modification?

Comment (andresailer) [reply include/ConformalTracking.h]: s/presents/prevents/ ?

File: src/ConformalTracking.cc [hunk 2]
```diff
@@ -1444,7 +1444,7 @@ void ConformalTracking::extendTracks(std::vector<KDTrack*>& conformalTracks, std
 
 // Extend seed cells
 void ConformalTracking::extendSeedCells(std::vector<Cell*>& cells, KDTree* nearestNeighbours, bool extendingTrack,
-                                        std::vector<KDCluster*> debugHits) {
+                                        const std::vector<KDCluster*>& /*debugHits*/) {
```
Comment (dhynds) [inline src/ConformalTracking.cc]: What does this notation /*debugHits*/ mean?

File: include/ConformalTracking.h (hunk 1 above)
Comment (dhynds) [reply include/ConformalTracking.h]: yes, should read prevents!

File: src/ConformalTracking.cc (hunk 2 above)
Comment (andresailer) [reply src/ConformalTracking.cc]: `/*debugHits*/` just comments the parameter name because it is unused

File: include/ConformalTracking.h (hunk 1 above)
Comment (andresailer) [reply include/ConformalTracking.h]: Then yes, it prevents modifications of the object

**Pull request merged**